File: LHDS.AdminPortal.Web/ClientApp/src/brokers/apiBroker.decryptions.ts
import ApiBroker from "./apiBroker";

class DecryptionBroker {
    relativeAuditUrl = '/api/decryptions';

    private apiBroker: ApiBroker = new ApiBroker();

    async GetDocumentByFileNameToDecryptAsync(fileName: string) {
        var url = this.relativeAuditUrl + '/' + encodeURIComponent(fileName);

        return await this.apiBroker.GetAsync(url);
    }
}
export default DecryptionBroker;

File: LHDS.AdminPortal.Api.Tests.Acceptance/Brokers/ApiBroker.Decryptions.cs
// ---------------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------------

using System.Threading.Tasks;

namespace LHDS.AdminPortal.Api.Tests.Acceptance.Brokers
{
    public partial class ApiBroker
    {
        private const string decryptionsRelativeUrl = "api/decryptions";

        public async ValueTask GetDocumentByFileNameToDecryptAsync(string fileName) =>
            await this.apiFactoryClient.GetContentStringAsync($"{decryptionsRelativeUrl}/{fileName}");
    }
}
